src/pages/swap/index.tsx

```diff
@@ -93,7 +108,7 @@ export const SwapPage = () => {
     tokenOutData.stableAmountInUnits != '' &&
     Big(tokenOutData.stableAmountInUnits).gt(Big(0));
 
-  function onSubmit(e: Event) {
+  function onConfirm(e: Event) {
```

Comment:
Since this function is quite large, let's consider defining it with `useCallback()`.

Reply:
Why would you prefer it inside a `useCallback` hook? Since it depends of most of the state UI elements it will re-render probably every time anyway. You think it could affect performance if not?

Reply:
Hmm right, it would have quite a few dependencies. I just thought caching with `useCallback` might make sense because of the calculations done in that function. But you're probably right that it wouldn't make much of a difference in this case.

Reply:
No strong opinion, just that it would make it even longer but we can do it.

Reply:
No need, let's just leave it as is.